packages/react-select/src/styles.ts

```diff
@@ -132,6 +108,27 @@ export const defaultStyles: StylesFunctions<
   valueContainer: valueContainerCSS,
 };
 
+export type StylesConfig<
+  Option = unknown,
+  IsMulti extends boolean = boolean,
+  Group extends GroupBase<Option> = GroupBase<Option>
+> = {
+  [K in keyof StylesProps<Option, IsMulti, Group>]?: (
+    base: CSSObjectWithLabel,
+    props: StylesProps<Option, IsMulti, Group>[K]
+  ) => CSSObjectWithLabel;
+};
+
+export type ClassNamesConfig<
+  Option,
+  IsMulti extends boolean,
+  Group extends GroupBase<Option>
```

Comment:
Should these be given default values like StylesConfig above?
```suggestion
	Option = unknown,
  IsMulti extends boolean = boolean,
  Group extends GroupBase<Option> = GroupBase<Option>
```

Reply:
Since we are exporting it now, it's probably better dx to. Feel free to commit your suggestion ✅